lib/core/src/Cardano/Wallet/Primitive/CoinSelection.hs

```diff
@@ -192,20 +217,19 @@ data SelectionParams = SelectionParams
     }
     deriving (Eq, Generic, Show)
 
--- | Indicates that an error occurred while performing a coin selection.
---
-data SelectionError
-    = SelectionBalanceError Balance.SelectionError
-    | SelectionOutputsError ErrPrepareOutputs
+-- | Indicates that coin selection failed, or a precondition to coin selection
+-- failed.
+data ErrWalletSelection
+    = ErrWalletSelectionOutputs ErrPrepareOutputs
+    | ErrWalletSelectionBalance SelectionError
     deriving (Eq, Show)
 
 -- | Prepares the given user-specified outputs, ensuring that they are valid.
---
-prepareOutputs
+validateTxOutsForSelection
```

Comment:
Suggestion: use `prepareTxOutsForSelection`.
Justification:
1. This function doesn't just perform validation: it also adjusts the contents of the outputs.
2. We've used "prepare" everywhere else. (`ErrPrepareOutputs` for example.)
```suggestion
prepareTxOutsForSelection
```